java.lang.NoSuchMethodError running j2obcXcode in Gradle 2.8 (Regression)

The android project is already correctly building and pulling in the shared component. Now I'm trying to build the xcode side. After running `pod init` in the ios directory, I run `gradle j2objcXcode` in android and get:

```
:shared:j2objcTranslatedMainLibraryExtraction UP-TO-DATE
:shared:compileJava UP-TO-DATE
:shared:processResources UP-TO-DATE
:shared:classes UP-TO-DATE
:shared:j2objcTranslatedTestLibraryExtraction UP-TO-DATE
:shared:compileTestJava UP-TO-DATE
:shared:processTestResources UP-TO-DATE
:shared:testClasses UP-TO-DATE
:shared:test UP-TO-DATE
:shared:j2objcPreBuild UP-TO-DATE
:shared:j2objcPodspec UP-TO-DATE
:shared:j2objcXcode FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':shared:j2objcXcode'.
> org.codehaus.groovy.runtime.StringGroovyMethods.replaceAll(Ljava/lang/CharSequence;Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Ljava/lang/CharSequence;

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':shared:j2objcXcode'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:69)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:35)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:64)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:53)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:203)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:185)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:62)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:50)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor.process(DefaultTaskPlanExecutor.java:25)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:110)
	at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:37)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
	at org.gradle.execution.DefaultBuildExecuter.access$000(DefaultBuildExecuter.java:23)
	at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:43)
	at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:32)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:30)
	at org.gradle.initialization.DefaultGradleLauncher$4.run(DefaultGradleLauncher.java:154)
	at org.gradle.internal.Factories$1.create(Factories.java:22)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:52)
	at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:151)
	at org.gradle.initialization.DefaultGradleLauncher.access$200(DefaultGradleLauncher.java:32)
	at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:99)
	at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:93)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:62)
	at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:93)
	at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:82)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:94)
	at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:43)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:28)
	at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:77)
	at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:47)
	at org.gradle.launcher.exec.DaemonUsageSuggestingBuildActionExecuter.execute(DaemonUsageSuggestingBuildActionExecuter.java:51)
	at org.gradle.launcher.exec.DaemonUsageSuggestingBuildActionExecuter.execute(DaemonUsageSuggestingBuildActionExecuter.java:28)
	at org.gradle.launcher.cli.RunBuildAction.run(RunBuildAction.java:43)
	at org.gradle.internal.Actions$RunnableActionAdapter.execute(Actions.java:170)
	at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:237)
	at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:210)
	at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:35)
	at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:24)
	at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:206)
	at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:169)
	at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:33)
	at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:22)
	at org.gradle.launcher.Main.doAction(Main.java:33)
	at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:45)
	at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:54)
	at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:35)
	at org.gradle.launcher.GradleMain.main(GradleMain.java:23)
Caused by: java.lang.NoSuchMethodError: org.codehaus.groovy.runtime.StringGroovyMethods.replaceAll(Ljava/lang/CharSequence;Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Ljava/lang/CharSequence;
	at com.github.j2objccontrib.j2objcgradle.tasks.XcodeTask$PodspecDetails.getPodMethodName(XcodeTask.groovy:106)
	at com.github.j2objccontrib.j2objcgradle.tasks.XcodeTask.podMethodLines(XcodeTask.groovy:487)
	at com.github.j2objccontrib.j2objcgradle.tasks.XcodeTask$_updatePodMethods_closure6.doCall(XcodeTask.groovy:453)
	at com.github.j2objccontrib.j2objcgradle.tasks.XcodeTask.updatePodMethods(XcodeTask.groovy:452)
	at com.github.j2objccontrib.j2objcgradle.tasks.XcodeTask.updatePodfile(XcodeTask.groovy:422)
	at com.github.j2objccontrib.j2objcgradle.tasks.XcodeTask.writeUpdatedPodfileIfNeeded(XcodeTask.groovy:385)
	at com.github.j2objccontrib.j2objcgradle.tasks.XcodeTask.xcodeConfig(XcodeTask.groovy:197)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.doExecute(AnnotationProcessingTaskFactory.java:227)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:220)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:209)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:585)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:568)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
	... 57 more
```

--- a/src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/TranslateTask.groovy
+++ b/src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/TranslateTask.groovy
@@ -165,14 +165,14 @@ class TranslateTask extends DefaultTask {
                 void execute(InputFileDetails details) {
                     // We must filter by srcFiles, since all possible input files are @InputFiles to this task.
                     if (originalMainSrcFiles.contains(details.file)) {
-                        logger.debug("New or Updated main file: " + details.file)
+                        getLogger().debug("New or Updated main file: " + details.file)
                         mainSrcFilesChanged += project.files(details.file)
                     } else if (originalTestSrcFiles.contains(details.file)) {
-                        logger.debug("New or Updated test file: " + details.file)
+                        getLogger().debug("New or Updated test file: " + details.file)
                         testSrcFilesChanged += project.files(details.file)
                     } else {
                         nonSourceFileChanged = true
-                        logger.debug("New or Updated non-source file: " + details.file)
+                        getLogger().debug("New or Updated non-source file: " + details.file)
                     }
                 }
             })
@@ -183,16 +183,16 @@ class TranslateTask extends DefaultTask {
                 void execute(InputFileDetails details) {
                     // We must filter by srcFiles, since all possible input files are @InputFiles to this task.
                     if (originalMainSrcFiles.contains(details.file)) {
-                        logger.debug("Removed main file: " + details.file.name)
+                        getLogger().debug("Removed main file: " + details.file.name)
                         String nameWithoutExt = details.file.name.toString().replaceFirst("\\..*", "")
                         removedMainFileNames += nameWithoutExt
                     } else if (originalTestSrcFiles.contains(details.file)) {
-                        logger.debug("Removed test file: " + details.file.name)
+                        getLogger().debug("Removed test file: " + details.file.name)
                         String nameWithoutExt = details.file.name.toString().replaceFirst("\\..*", "")
                         removedTestFileNames += nameWithoutExt
                     } else {
                         nonSourceFileChanged = true
-                        logger.debug("Removed non-source file: " + details.file)
+                        getLogger().debug("Removed non-source file: " + details.file)
                     }
                 }
             })

--- a/src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/Utils.groovy
+++ b/src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/Utils.groovy
@@ -413,7 +413,7 @@ class Utils {
     static FileTree javaTrees(Project proj, List<String> treePaths) {
         List<? extends FileTree> trees =
             treePaths.collect({ String treePath -> proj.fileTree(dir: treePath, includes: ["**/*.java"]) })
-        return new UnionFileTree("javaTrees_j2objc", trees)
+        return new UnionFileTree("javaTrees_j2objc", (Collection<? extends FileTree>) trees)
     }
 
     static List<String> j2objcLibs(String j2objcHome,

--- a/src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/XcodeTask.groovy
+++ b/src/main/groovy/com/github/j2objccontrib/j2objcgradle/tasks/XcodeTask.groovy
@@ -102,7 +102,7 @@ class XcodeTask extends DefaultTask {
 
         String getPodMethodName() {
             // Valid Ruby name requires replacing all non-alphanumeric characters with underscore
-            return "j2objc_$projectName".replaceAll(/[^a-zA-Z0-9]/, '_')
+            return "j2objc_$projectName".toString().replaceAll(/[^a-zA-Z0-9]/, '_')
         }
 
         @SuppressWarnings(['unused', 'grvy:org.codenarc.rule.unused.UnusedPrivateMethodRule'])

--- a/src/test/groovy/com/github/j2objccontrib/j2objcgradle/J2objcConfigTest.groovy
+++ b/src/test/groovy/com/github/j2objccontrib/j2objcgradle/J2objcConfigTest.groovy
@@ -255,6 +255,8 @@ class J2objcConfigTest {
     // A small number of the configuration variable must be String[]
     // instead of List<String>, this tests 'extraLinkerArgs' as an example.
     @Test
+    // Skip to allow: "= ['-arg1']" as it would be used in j2objcConfig
+    @CompileStatic(TypeCheckingMode.SKIP)
     void testStringArrayArgs() {
         J2objcConfig j2objcConfig = new J2objcConfig(proj)
         j2objcConfig.extraLinkerArgs = ['-arg1']
@@ -282,6 +284,8 @@ class J2objcConfigTest {
     }
 
     @Test
+    // Skip to allow: "+= 'ios_armv7s'" as it would be used in j2objcConfig
+    @CompileStatic(TypeCheckingMode.SKIP)
     void testSupportedAndEnabledArchs_SubsetEnabledArchsSpecifiedWithAdditionalSupportedArchs() {
         J2objcConfig j2objcConfig = TestingUtils.setupProjectJ2objcConfig(
                 new TestingUtils.ProjectConfig(createJ2objcConfig: true,

